madara/crates/client/rpc/src/versions/user/v0_7_1/methods/read/syncing.rs

```diff
@@ -16,30 +16,63 @@ use crate::Starknet;
 /// * `Syncing` - An Enum that can either be a `mc_rpc_core::SyncStatus` struct representing the
 ///   sync status, or a `Boolean` (`false`) indicating that the node is not currently synchronizing.
 pub async fn syncing(starknet: &Starknet) -> StarknetRpcResult<SyncingStatus> {
-    // obtain best seen (highest) block number
+    // Get the sync status from the provider
+    let sync_status = starknet.sync_status().await.or_internal_server_error("Error getting sync status")?;
+
+    // Get current block info
     let Some(current_block_info) = starknet
         .backend
         .get_block_info(&BlockId::Tag(BlockTag::Latest))
         .or_internal_server_error("Error getting latest block")?
     else {
-        return Ok(SyncingStatus::NotSyncing); // TODO: This doesn't really make sense? This can only happen when there are no block in the db at all.
+        return Ok(SyncingStatus::NotSyncing);
     };
 
     let current_block_info =
         current_block_info.as_nonpending().ok_or_internal_server_error("Latest block cannot be pending")?;
-    let starting_block_num = 0; // TODO(rpc): fix this // starknet.starting_block;
-    let starting_block_info = starknet.get_block_info(&BlockId::Number(starting_block_num))?;
-    let starting_block_info =
-        starting_block_info.as_nonpending().ok_or_internal_server_error("Block cannot be pending")?;
-    let starting_block_hash = starting_block_info.block_hash;
     let current_block_num = current_block_info.header.block_number;
     let current_block_hash = current_block_info.block_hash;
 
+    // Get the starting block number from sync status
+    let starting_block_num = sync_status.starting_block_num;
+
+    // Get the starting block hash - if it's not in sync status or is zero, fetch it from the backend
+    let starting_block_hash = if sync_status.starting_block_hash == BlockHash::default() {
+        // We need to fetch the starting block hash from the backend
+        let Some(starting_block_info) = starknet
+            .backend
+            .get_block_info(&BlockId::Number(starting_block_num))
+            .or_internal_server_error("Error getting starting block")?
+        else {
+            // If we can't get the starting block info, use a default hash
+            // This is a fallback and shouldn't normally happen
+            tracing::warn!("Could not get starting block info for block {}", starting_block_num);
+            return Ok(SyncingStatus::NotSyncing);
+        };
+
+        let starting_block_info =
+            starting_block_info.as_nonpending().ok_or_internal_server_error("Starting block cannot be pending")?;
+
+        starting_block_info.block_hash
```

Comment:
Also here should update `starting_block_num`, just to be sure that `starting_block_hash` and `starting_block_num` refers to the same block.

Reply:
yep, you're right. No need to update it 😃